geopandas/base.py

```diff
@@ -24,11 +24,11 @@ def is_geometry_type(data):
         return False
 
 
-def _delegate_binary_method(op, this, other, *args, **kwargs):
+def _delegate_binary_method(op, this, other, align, *args, **kwargs):
     # type: (str, GeoSeries, GeoSeries) -> GeoSeries/Series
     this = this.geometry
     if isinstance(other, GeoPandasBase):
-        if not this.index.equals(other.index):
+        if align and not this.index.equals(other.index):
```

Comment:
```suggestion
        if align is None and not this.index.equals(other.index):
```

Idea: make the default `None`, which is treated as True but gives a warning when alignment is actually done. This gives a way for the user to indicate they are OK with alignment without getting warnings about it. 
(small disadvantage is that the signature in the function is less explicit than with `align=True`)

Reply:
I do understand your point, but considering how troublesome this behaviour is (I responded to countless questions on this), I would prefer having an explicit `align=True` in the signature. Having `None/True/False` is a nice trick for advanced users to silence warnings, but in this specific case, I would prefer understanding for basic users (I really don't want to answer the same question again :D).

Reply:
Then we should maybe rather make it an error by default? To make it really clear for users they are doing something wrong (with a good explanation then) 
But happy to leave this as a follow-up discussion to not block this PR ;)

Reply:
That'd be a breaking change.

> But happy to leave this as a follow-up discussion to not block this PR ;)

I'll open an issue as a follow-up so we can get closer to 0.9 :).

Reply:
#1832